### PR TITLE
[MINOR] Avoid passing the PermGenSize option to IBM JVMs.

### PR DESCRIPTION
**Pull request by tellison**
IBM's Java VM doesn't have the concept of a permgen, so this option shouldn't be passed when the vendor property shows it is an IBM JDK.

### CODE REVIEW

**Comment by srowen** [comment]
Does it actually fail to start if it sees this option?

File: launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java [hunk 1]
```diff
@@ -108,6 +113,23 @@ static boolean isWindows() {
     return os.startsWith("Windows");
   }
 
+  /**  Returns an enum value indicating whose JVM is being used. */
+  static JavaVendor getJavaVendor() {
+    String vendorString = System.getProperty("java.vendor");
+    if (vendorString.contains("Oracle")) {
+      return JavaVendor.Oracle;
+    } else {
```
Comment (srowen) [inline launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java]: Minor: This doesn't have to have the `else`s and thus extra level or two of nesting.

**Comment by srowen** [comment]
Jenkins, add to whitelist

**Comment by tellison** [comment]
The IBM VM will still start if passed -XX options that have no impact, so this is just being a good citizen in case that position changes in the future.

There are other IBM JVM changes I have that are a bit more significant.  Let me know if you want an umbrella JIRA to cover running on IBM Java.

File: launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java (hunk 1 above)
Comment (tellison) [inline launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java]: ACK

**Comment by srowen** [comment]
My only concern is complicating the code for args that don't actually cause a problem; I imagine any JVM should often be prepared to ignore `-XX` args it doesn't know since they are such a free-for-all. Still, not much harm in this particular change. But is there a lot more you have in mind?

**Comment by tellison** [comment]
Not for -XX args.  I have some changes to SizeEstimator, endianness, etc. for correctness, and thinking of extending java-opts to allow for vendor specific args.  We can discuss on dev list and/or JIRA.

**Comment by srowen** [comment]
Looks OK to me, excepting the minor simplification of those else branches.

**Pull request closed without merging**